### ISSUE TITLE
struct2depth Batch_normalization set to training mode during evaluation?

### ISSUE THREAD

**Issue by aj96** [tensorflow/models#6927]
### System information
- **What is the top-level directory of the model you are using**: struct2depth
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **TensorFlow installed from (source or binary)**: pip install tensorflow-gpu
- **TensorFlow version (use command below)**: 1.13
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: 10.0 / 7.4.1
- **GPU model and memory**: Tesla P-100 16Gb
- **Exact command to reproduce**: Run evaluation script on researchers' model

Why is it that in the build_test_depth_graph() function in model.py the is_training flag is hardcoded to True? This causes all the batch normalization layers to be in training mode, which is causing the evaluation to be inconsistent when inferencing with different batch sizes. 

However, when I do change this to be False, the evaluation metrics become terrible. I am using the researchers' provided pre-trained model.

When evaluating in training mode with batch size=4, I got the following metrics:
   abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.1455,     1.1230,     5.3762,     0.2183,     0.0000,     0.8131,     0.9428,     0.9777

When evaluating in inference mode, I get the following metrics:
   abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.4708,     5.3870,    12.7005,     0.8727,     0.0000,     0.2708,     0.5032,     0.6897

Can anyone explain this? 

As a final note, I am using the evaluation code from the original SfMLeaner repo as the evaluation script was not provided in the original struct2depth repo.

@aneliaangelova Can you please comment on this? According to the tensorflow documentation, you must explicitly add the update_ops to properly train batch normalization. I even tried adding this and trained another model, but the results are still terrible when inferencing with the is_training flag set to false.

Documentation link: https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization

**Comment by aneliaangelova** [comment]
Yes we trained and tested with is_traininig = True, so the person posting this is correct. The code reflects what we trained and tested with correctly. We welcome improvements to the code with batch_normalization.

**Comment by aj96** [comment]
Thank you so much for your reply. I was mostly interested in whether you had an explanation for such weird behavior?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
